aiida/engine/processes/calcjobs/calcjob.py

```diff
@@ -315,17 +291,27 @@ def presubmit(self, folder):
         computer = self.node.computer
         inputs = self.node.get_incoming(link_type=LinkType.INPUT_CALC)
 
+        if not self.inputs.metadata.dry_run and self.node.has_cached_links():
+            raise InvalidOperation('calculation node has unstored links in cache')
+
         codes = [_ for _ in inputs.all_nodes() if isinstance(_, Code)]
 
-        calcinfo = self.prepare_for_submission(folder)
-        scheduler = computer.get_scheduler()
+        for code in codes:
+            if not code.can_run_on(computer):
```

Comment:
is this the equivalent of the previous `is_local`/`get_local_executable`/`get_content_list` check?

Reply:
It is a different check, this one was just done in the `run` method while the one you reference was being done in `presubmit`. I just moved the former here as it makes more sense here. I realize now I can merge the two loops. I will do that

Reply:
@muhrin I merged the two checks in the same loop, so this should be good to go